File: leetcode-in-python/solution/9_palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        n = x
        r = 0
        # negative number will not reverse
        while n > 0:
           r = r * 10 + n % 10
           n //= 10
        return x == r


import unittest


class TestSolution(unittest.TestCase):

    def setUp(self):
        self.input = [123, 121, -121, -0, 0, 1221, 12321]
        self.answer = [False, True, False, True, True, True, True]
        self.s = Solution()

    def test_solution(self):
        for i, item in enumerate(self.input):
            ans = self.s.isPalindrome(item)
            self.assertEqual(ans , self.answer[i],ans )


if __name__ == '__main__':
    unittest.main()

File: leetcode-in-rust/src/solution/_9_palindrome_number.rs
pub struct Solution;

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let mut n = x;
        let mut r = 0;
        while n > 0 {
            r = r * 10 + n % 10;
            n /= 10;
        }
        x == r
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_solution() {
        let test_case = vec![
            (123, false),
            (121, true),
            (-121, false),
            (-0, true),
            (0, true),
            (1221, true),
            (12321, true),
        ];

        for (x, answer) in test_case{
            let ans = Solution::is_palindrome(x);
            assert_eq!(ans, answer);
        }
    }
}
